### PR TITLE
solution Method join

### PR DESCRIPTION
**Pull request by Zelinskyi-Serhii**

### CODE REVIEW

File: src/arrayMethodJoin.js [hunk 1]
```diff
@@ -5,7 +5,30 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let between = separator;
+
+    if (separator === undefined) {
+      between = ',';
+    }
```
Comment (rashyd-hasratov) [inline src/arrayMethodJoin.js:13]: Actually, you don't need this part of code, you can just set the default value for separator in the function params list.

File: src/arrayMethodJoin.js [hunk 2]
```diff
@@ -5,7 +5,30 @@
...
+        continue;
+      }
+
+      if (i + 1 !== this.length) {
+        result += `${this[i]}${between}`;
+        continue;
+      }
+
+      if (this[this.length - 1] !== undefined) {
+        result += this[i];
```
Comment (rashyd-hasratov) [inline src/arrayMethodJoin.js:27]: I would like to offer you such simplification for this part of code:
1) if it is not the first element, add separator to result
2) if it is null or undefined, just skip this iteration
3) else add the element to the result

**Review by rashyd-hasratov — CHANGES_REQUESTED** [review]
Nice, but it is possible to make the solution even better!

File: src/arrayMethodJoin.js [hunk 3]
```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const between = separator;
```
Comment (OleksiiBlezniuk) [inline src/arrayMethodJoin.js:9]: But why do you need this constant? You already have `separator`

**Review by OleksiiBlezniuk — CHANGES_REQUESTED** [review]
Well done, but let's avoid using unnecessary constants

**Review by OleksiiBlezniuk — APPROVED** [review]
Well done!

**Pull request closed without merging**